### ISSUE TITLE
[Bug]: Windows Server 2019 Memory Leak

### ISSUE THREAD

**Issue by M00T1CU5** [netdata/netdata#19212]
### Bug description

Over several days the netdata agent on Windows Server 2019 will continue using more and more memory. Currently, 2 days after the last service restart it's using 500mb of memory. I've seen it get up to 1.5GB memory usage. I have to restart the service regularly to get it back down to normal memory usage ~100mb. I am running v2.0.3. I also have the client running on a Windows 10 install, which does not experience this same issue. 

### Expected behavior

Memory usage stays constant, and at a lower amount.

### Steps to reproduce

1. Install client on Windows Server 2019
2. Leave client run for several days
3. Check memory usage

### Installation method

windows MSI

### System info

```shell
OS Name:                   Microsoft Windows Server 2019 Standard
OS Version:                10.0.17763 N/A Build 17763
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Server
OS Build Type:             Multiprocessor Free
```

### Netdata build info

```shell
v2.0.3
```

**Comment by thiagoftsm** [comment]
Hello @M00T1CU5 ,

Thank you for your report. And we apologize for the issues.

To help us accurately recreate the scenario and properly address the issue, I have a few questions for you:

1. Are you using the default configuration?
2. Is either SQL Server or IIS running on this host?
3. Is this host receiving metrics from another host?

Looking forward to your response.

Best regards,

**Comment by M00T1CU5** [comment]
Hello,

1. Are you using the default configuration? **Yes**
2. Is either SQL Server or IIS running on this host? **No**
3. Is this host receiving metrics from another host? **No**

I've attached a screenshot from today, which shows the service is using 2.6GB of RAM. Currently running v2.1
![2025-01-11_103227_Neptune_-_ Base_Host__Windows_Server_2019 _-_Royal](https://github.com/user-attachments/assets/9c7a94a9-54f2-4493-83d3-d22be7d10db0)

**Comment by thiagoftsm** [comment]
First, I apologize for the delay in responding. I only recently gained access to a Windows 2019 server, which I am now using to attempt to recreate the issue you described.

On our Windows 2022 server, I have not observed the behavior you mentioned. To better align the server I’m working on with your environment, I’d like to ask a few additional questions:

- Are you running the latest version of Netdata? Is this issue still occurring?
- We recently merged new features, such as Active Directory monitoring. Is your server running Active Directory?

Thank you for your patience, and I look forward to your response.

Best regards

**Comment by M00T1CU5** [comment]
Thanks for your response, no worries, it's kinda minor - I just have a scheduled task to run to restart the service hourly. 

- I have just installed the latest version, and have disabled the restart task so i will check in in a couple days time to let you know how the memory usage is.
- Not running Active Directory services. The only role/feature i've added to the server is Hyper-V.

**Comment by thiagoftsm** [comment]
Thank you very much for your responses.

It looks like I was able to replicate the issues you reported:

![Image](https://github.com/user-attachments/assets/fa2754d1-0ce8-4cdb-9ebe-b8d40e25b1e1)

Additionally, I am currently using this host to enhance our Hyper-V collector.

I will provide updates as soon as possible.

Best regards!

**Comment by ilyam8** [comment]
> I was able to replicate the issues you reported

Do you mean this 142->152 MB memory increase? Or the wrong screenshot?

**Comment by ilyam8** [comment]
No, it isn't consistent with yours. Characterizing a 10MB increase over several hours after startup as a 'reproduction' of the issue may be premature.

**Comment by thiagoftsm** [comment]
Hello @M00T1CU5 ,

Since our last conversation, we’ve been investigating the issue and have identified some problems. @stelfrag  and I are actively working on fixing them.

![Image](https://github.com/user-attachments/assets/ac5c4dd5-1fbe-47f8-9f70-81fa70ce8f6a)

**Comment by M00T1CU5** [comment]
Hi @thiagoftsm ,

Thanks for the update! If you need me to test anything, be sure to let me know. :)

**Comment by M00T1CU5** [comment]
Since updating my server to Windows Server 2025, i am not longer having this issue. Seems like it was specific to Windows Server 2019.

**Comment by thiagoftsm** [comment]
> Since updating my server to Windows Server 2025, i am not longer having this issue. Seems like it was specific to Windows Server 2019.

Hello @M00T1CU5 ,

We've conducted tests using the same installer across Windows Server 2019, 2022, and 2025 environments. Consistent with your findings, we've only observed these specific events occurring on Windows Server 2019.

While we're still investigating behaviors in older Windows Server versions, we currently don't have a conclusive explanation for why these events appear exclusively in the 2019 environment.

We appreciate your update and will keep you informed as we uncover more information.
